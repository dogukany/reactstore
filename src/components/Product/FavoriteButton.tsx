import { Ionicons } from "@expo/vector-icons";
import { TouchableOpacity } from "react-native";
import styles from "./FavoriteButton.styles";

interface FavoriteButtonProps {
  isFavorite: boolean;
  handleFavorite: () => void;
}

const FavoriteButton = ({
  isFavorite,
  handleFavorite,
}: FavoriteButtonProps) => {
  return (
    <TouchableOpacity style={styles.container} onPress={handleFavorite}>
      <Ionicons
        name={isFavorite ? "heart" : "heart-outline"}
        size={30}
        color={isFavorite ? "red" : "black"}
      />
    </TouchableOpacity>
  );
};
export default FavoriteButton;
