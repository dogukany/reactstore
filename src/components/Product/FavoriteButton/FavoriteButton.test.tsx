import renderer from "react-test-renderer";

import FavoriteButton from "./FavoriteButton";

describe("FavoriteButton", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<FavoriteButton isFavorite handleFavorite={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
