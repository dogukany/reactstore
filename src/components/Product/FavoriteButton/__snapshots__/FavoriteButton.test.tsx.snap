// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteButton renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
      "position": "absolute",
      "right": 10,
      "top": 5,
      "zIndex": 1,
    }
  }
>
  <Text />
</View>
`;
